vterebey/budgetcalculator/budget_calculator_class.py

```diff
@@ -0,0 +1,302 @@
+    return user_category
+
+
+def _input_valid():
+    """This fucntion validates input menu data"""
+    input_valid = False
+    while not input_valid:
+        try:
+            user_choice = int(input("> "))
+            input_valid = True
+        except ValueError:
+            print("Only integers ale allowed, repeat input\n")
+            input_valid = False
+    return user_choice
+
+
+def _select_valid():
+    """This function validates input category from default categories"""
+    select_valid = False
+    while not select_valid:
+        try:
+            user_select = int(input("Enter number of category:"))
+            if 1 <= user_select <= 6:
+                select_valid = True
+            else:
+                print("Only integers from 1 to 6, repeat input\n")
+        except ValueError:
+            print("Only integers from 1 to 6, repeat input\n")
+            select_valid = False
+    return user_select
+
+
+def _enter_valid():
+    """This function validates and return category"""
+    enter_valid = False
+    while not enter_valid:
+        try:
+            user_cat = input("Enter Category: ")
+            if cat_pattern.search(user_cat) is None:
+                enter_valid = True
+            else:
+                print("Repeat input\n")
+        except ValueError:
+            print("Repeat input\n")
+            enter_valid = False
+    return user_cat
+
+
+def enter_month():
+    """This function validates input category"""
+    enter_valid = False
+    while not enter_valid:
+        user_date = input("Enter Month(YYYY-MM): ")
+        try:
+            valid_date = datetime.strptime(user_date, "%Y-%m")
+            valid_date = valid_date.strftime("%Y-%m")
+            enter_valid = True
+        except ValueError:
+            print("Invalid date! Repeat input ")
+            enter_valid = False
+    return valid_date
+
+
+def enter_amount():
+    """This function validates enters amount"""
+    amount_valid = False
+    while not amount_valid:
+        try:
+            user_amount = float(input("Enter Amount: "))
+            if user_amount > 0:
+                amount_valid = True
+                print()
+            else:
+                print("Only numbers greater than 0. Come on!\n")
+        except ValueError:
+            print("Only numbers greater than 0. Come on!\n")
+    return user_amount
+
+
+def enter_date():
+    """This function enters date"""
+    enter_valid = False
+    while not enter_valid:
+        user_date = input("Enter Date(YYYY-MM-DD): ")
+        try:
+            valid_date = datetime.strptime(user_date, "%Y-%m-%d")
+            valid_date = valid_date.date()
+            enter_valid = True
+        except ValueError:
+            print("Invalid date! Repeat input ")
+            enter_valid = False
+    return valid_date
+
+
+def main():
+    """This function executes the program"""
+    while True:
+        welcome_menu()
+        choice = _input_valid()
```

Comment:
there should be 2 different functions: for user input and for validation
```
user_input = get_user_input()
if _is_user_input_valid(user_input ):
 ......
```

Reply:
Updated